lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -123,5 +126,35 @@ def find_latest_rs(deployment_data)
       rs.sync(latest_rs_data)
       rs
     end
+
+    def min_unavailable_rollout
+      @definition.dig('metadata', 'annotations', 'kubernetes-deploy.shopify.io/required-rollout')
+    end
+
+    def minimum_unavailable_replicas_to_succeeded
+      desired = @desired_replicas
+
+      case min_unavailable_rollout
+      when 'maxUnavailable'
+        max_unavailable = @definition.dig('spec', 'strategy', 'rollingUpdate', 'maxUnavailable')
+        if max_unavailable =~ /%/
+          (desired * (100 - max_unavailable.to_i) / 100.0).to_i
```

Comment:
This won't match k8s behaviour, because k8s rounds up (well, it rounds the maxUnavailable num down) and `Float.to_i` simply truncates (i.e. rounds down).

e.g. desired = 5, maxUnavailable = 25%, we should return 4 here and we'll return 3

Reply:
I know there's not much in the way of infrastructure for this, but we should probably get some unit tests on this logic.